frontend/src/components/CustomTable.test.tsx

```diff
@@ -420,4 +536,40 @@ describe('CustomTable', () => {
     await TestUtils.flushPromises();
     expect(tree).toMatchSnapshot();
   });
+
```

Comment:
I added three snapshot tests up around line 74. Do you mean in addition to those?

Reply:
Sorry, should've made a clearer comment. Yes, it seemed like handleFilterChange isn't tested below because of the debounce issue, so perhaps a snapshot test will make sure that typing something eventually makes it appear in the DOM?

Reply:
The debounce issue doesn't affect that, since it's the `setStateSafe` call that updates the UI (the debounce only covers the API call). The only part that isn't tested is the connection between `handleFilterChange` and `_requestFilter`.

We test whether `handleFilterChange` updates the `filterString` value of the `state`, and that `_requestFilter` encodes the filter and reloads the page. 

Still, I don't think there's harm in adding a snapshot check in the `handleFilterChange` to confirm that the UI is being updated.

Reply:
I see, makes sense. Cool, thanks for adding anyway!